src/test/regress/sql/insert_select_into_local_table.sql

```diff
@@ -0,0 +1,99 @@
+CREATE TABLE dist_table (a INT, b INT, c TEXT);
+SELECT create_distributed_table('dist_table', 'a');
+
+INSERT INTO dist_table VALUES (1, 6, 'txt1'), (2, 7, 'txt2'), (3, 8, 'txt3');
+
+CREATE TABLE non_dist_1 (a INT, b INT, c TEXT);
+CREATE TABLE non_dist_2 (a INT, c TEXT);
+CREATE TABLE non_dist_3 (a INT);
+
+
+-- test non-router queries
+INSERT INTO non_dist_1 SELECT * FROM dist_table;
+INSERT INTO non_dist_2 SELECT a, c FROM dist_table;
+INSERT INTO non_dist_3 SELECT a FROM dist_table;
+
+SELECT * FROM non_dist_1 ORDER BY 1, 2, 3;
+SELECT * FROM non_dist_2 ORDER BY 1, 2;
+SELECT * FROM non_dist_3 ORDER BY 1;
+
+TRUNCATE non_dist_1, non_dist_2, non_dist_3;
+
+
+-- test router queries
+INSERT INTO non_dist_1 SELECT * FROM dist_table WHERE a = 1;
+INSERT INTO non_dist_2 SELECT a, c FROM dist_table WHERE a = 1;
+INSERT INTO non_dist_3 SELECT a FROM dist_table WHERE a = 1;
+
+SELECT * FROM non_dist_1 ORDER BY 1, 2, 3;
+SELECT * FROM non_dist_2 ORDER BY 1, 2;
+SELECT * FROM non_dist_3 ORDER BY 1;
+
+TRUNCATE non_dist_1, non_dist_2, non_dist_3;
+
+
+-- test columns in different order
+INSERT INTO non_dist_1(b, a, c) SELECT a, b, c FROM dist_table;
+SELECT * FROM non_dist_1 ORDER BY 1, 2, 3;
+
+TRUNCATE non_dist_1;
+
+
+-- test EXPLAIN
+EXPLAIN INSERT INTO non_dist_1 SELECT * FROM dist_table;
+EXPLAIN INSERT INTO non_dist_1 SELECT * FROM dist_table WHERE a = 1;
+
+
+-- test RETURNING
+INSERT INTO non_dist_1 SELECT * FROM dist_table RETURNING *;
```

Comment:
This should have an ORDER BY somewhere:

```suggestion
INSERT INTO non_dist_1 SELECT * FROM dist_table ORDER BY 1, 2, 3 RETURNING *;
```